Unknown language: "jade"

Highlight.js does not support Jade Syntax. This breaks the build

> /Users/axawinterthur/dev/style-guide/node_modules/metalsmith-jade/node_modules/jade/lib/runtime.js:202
>   throw err;
>   ^
> 
> Error: /Users/axawinterthur/dev/style-guide/docs/page/changelog.jade:14
>     12|           .changelog__timeline__item__version
>     13|             span= item.name
>   > 14|             if item.published_at
>     15|               - var date = moment(item.published_at);
>     16|               .changelog__timeline__item__version__date= date.format('ll')
>     17|             else
> 
> Unknown language: "jade"
> Please report this to https://github.com/chjj/marked.
>     at Object.highlight (/Users/axawinterthur/dev/style-guide/node_modules/highlight.js/lib/highlight.js:443:13)
>     at Object.highlight (tasks/docs-pages.js:52:24)
>     at Object.renderer.code (lib/marked-renderer.js:134:24)
>     at Parser.tok (/Users/axawinterthur/dev/style-guide/node_modules/marked/lib/marked.js:979:28)
>     at Parser.tok (/Users/axawinterthur/dev/style-guide/node_modules/marked/lib/marked.js:1058:22)
>     at Parser.tok (/Users/axawinterthur/dev/style-guide/node_modules/marked/lib/marked.js:1038:22)
>     at Parser.parse (/Users/axawinterthur/dev/style-guide/node_modules/marked/lib/marked.js:924:17)
>     at Function.Parser.parse (/Users/axawinterthur/dev/style-guide/node_modules/marked/lib/marked.js:911:17)
>     at marked (/Users/axawinterthur/dev/style-guide/node_modules/marked/lib/marked.js:1213:19)
>     at eval (eval at <anonymous> (/Users/axawinterthur/dev/style-guide/node_modules/metalsmith-jade/node_modules/jade/lib/jade.js:172:8), <anonymous>:77:97)
> 
> npm ERR! Darwin 15.5.0
> npm ERR! argv "/Users/axawinterthur/.nvm/versions/node/v5.10.1/bin/node" "/Users/axawinterthur/.nvm/versions/node/v5.10.1/bin/npm" "run" "build"
> npm ERR! node v5.10.1
> npm ERR! npm  v3.8.3
> npm ERR! code ELIFECYCLE
> npm ERR! @axa-ch/style-guide@0.14.0 build: `gulp build`
> npm ERR! Exit status 1
> npm ERR!
> npm ERR! Failed at the @axa-ch/style-guide@0.14.0 build script 'gulp build'.
> npm ERR! Make sure you have the latest version of node.js and npm installed.
> npm ERR! If you do, this is most likely a problem with the @axa-ch/style-guide package,
> npm ERR! not with npm itself.
> npm ERR! Tell the author that this fails on your system:
> npm ERR!     gulp build
> npm ERR! You can get information on how to open an issue for this project with:
> npm ERR!     npm bugs @axa-ch/style-guide
> npm ERR! Or if that isn't available, you can get their info via:
> npm ERR!     npm owner ls @axa-ch/style-guide
> npm ERR! There is likely additional logging output above.
> 
> npm ERR! Please include the following file with any support request:
> npm ERR!     /Users/axawinterthur/dev/style-guide/npm-debug.log

--- a/lib/marked-renderer.js
+++ b/lib/marked-renderer.js
@@ -120,22 +120,17 @@ renderer.listitem = function (text) {
 }
 
 renderer.code = function (code, lang) {
-  let out = code
+  const highlight = this.options.highlight
+  let out = highlight(code, lang)
 
-  if (!lang) {
+  if (!lang || !out) {
+    out = code
+
     return `<pre class="highlight">
         <code class="highlight__listing hljs">
           ${out}
         </code>
       </pre>`
-  }
-
-  if (this.options.highlight) {
-    out = this.options.highlight(code, lang)
-
-    if (out === null) {
-      out = code
-    }
   }
 
   return `<pre class="highlight">

--- a/tasks/docs-pages.js
+++ b/tasks/docs-pages.js
@@ -49,7 +49,7 @@ module.exports = (cb) => {
     renderer: markedRenderer,
     langPrefix: '',
     highlight(code, lang) {
-      return highlight.highlight(lang, code).value
+      return highlight.getLanguage(lang) ? highlight.highlight(lang, code).value : undefined
     },
   })
 

